src/frontend/js/components/AshleyEditor/index.spec.tsx

```diff
@@ -29,8 +35,81 @@ describe('AshleyEditor', () => {
 
     // check if link button has been added to the toolbar plugin editor
     const linkButton = container.querySelector(
-      'path[d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"]',
-    );
+      'path[d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"]');
+
     expect(linkButton).toBeInTheDocument();
+
+    // check if code button has been added to the toolbar plugin editor
+    const codeButton = container.querySelector(
+      'path[d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"]',
+    );
+    expect(codeButton).toBeInTheDocument();
+    
+  });
+
+  it('loads ashley component with existing json content containing a code-block and an unstyled block',() => {
+    const target = document.createElement('input');
+    target.value = '{"blocks":[{"key":"brt6g","text":"ceci est un bloc code\n","type":"code-block","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"fh6un","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"6an2m","text":"ceci est une phrase en bold","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":0,"length":27,"style":"BOLD"}],"entityRanges":[],"data":{}}],"entityMap":{}}';
+    const { container } = render(<AshleyEditor target={target} />);
+    // check that the content is now present in the editor
+    screen.getByText(/ceci est un bloc code/i)
```

Comment:
I wonder if it wouldn't be better to also test Component Instance in this case, to test directly modification on the block map ?
For example, you can access to the component instance with this library [react-dom-instance](https://www.npmjs.com/package/react-dom-instance).

Reply:
The component we are testing is Ashley and its job is to render the editor state in the input. Wouldn't it be testing draftjs component rather than Ashley to try to get to these data. I'm not quiet sur I'll succeed to be honest but apart from that I think to test the value of the input is fare according to the role of our component.

Reply:
I think we have to ensure that the right html entity is used for its related code block. In this case this is a `pre`.

Just mention explicitly the expected html entity in the selector should do the job finally.
```js
expect(container.querySelector('pre.public-DraftStyleDefault-pre [data-text="true"]'))
  .toHaveTextContent('ce texte est ajouté dans code-block.');
```